### ISSUE TITLE
CompareWorkspaces crashes

### ISSUE THREAD

**Issue by gemmaguest** [mantidproject/mantid#22345]
I am seeing a crash with `CompareWorkspaces` in the nightly build (found while testing #21742)

To reproduce, run this script (you'll need to change the hard-coded save and load paths):

```
# This script produces a crash in mantid nightly 3.12.20180426.1630 on linux.
# I've also seen the following error sometimes in relation to this:
#
#  CompareWorkspaces: vector::_M_range_check: __n (which is 1) >= this->size() (which is 1)
#  Error in execution of algorithm CompareWorkspaces:
#  CompareWorkspaces: error (see log)
#
# Sometimes I get the above error without a crash and sometimes it works. It seems to be a timing
# issue and depends a bit on whether I execute the whole script together or as separate lines.

LoadISISNexus(Filename='INTER00043583.nxs', OutputWorkspace='TOF_43583')
ReflectometryReductionOne(InputWorkspace='TOF_43583', ProcessingInstructions='45-69', WavelengthMin='1', WavelengthMax='12', OutputWorkspace='IvsQ', OutputWorkspaceWavelength='IvsLam')
SaveNexus(InputWorkspace='IvsQ', Filename='/home/gemma/work/data/mantid/IvsQ_saved.nxs')
Load(Filename='/home/gemma/work/data/mantid/IvsQ_saved.nxs', OutputWorkspace='IvsQ_saved')
ReflectometryReductionOne(InputWorkspace='TOF_43583', ProcessingInstructions='45-69', WavelengthMin=1, WavelengthMax=12, OutputWorkspace='IvsQ', OutputWorkspaceWavelength='IvsLam')
CompareWorkspaces(Workspace1='IvsQ_saved', Workspace2='IvsQ', Tolerance=1.0000000000000001e-05, CheckInstrument=False, CheckSample=True)
```

### Platforms affected
Ubuntu 16.04

**Comment by keeeto** [comment]
I am on holiday until 8th, but happy to look at this when I get back!

**Comment by keeeto** [comment]
Hi @gemmaguest, I have only got around to this now, sorry! The above script seems to run fine on OSX and Windows, with no crashes. Possibly an issue only with Mantid on Ubuntu/Linux??

**Comment by keeeto** [comment]
Wait, actually my nightly builds were out of date, I reproduce the crash on Windows now. Rebuilding and checking OSX.

**Comment by keeeto** [comment]
Funny, the script runs fine on OSX, but not Windows.

**Comment by gemmaguest** [comment]
This seems to be fixed in v6.0.0
